convert/lib/nuke.py

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python3
+import re
+from functools import cached_property
+from pprint import pprint
+
+import boto3
+
+from .meta import cdk_ids
+
+
+class nuke:
+    def __init__(self, region: str, verbose: bool = False, delete: bool = False):
+        self.region = region
+        self.verbose = verbose
+        self.delete = delete
+
+    @cached_property
+    def autoscaling(self):
+        return boto3.client("autoscaling", self.region)
+
+    @cached_property
+    def ec2(self):
+        return boto3.client("ec2", self.region)
+
+    @cached_property
+    def iam(self):
+        return boto3.client("iam", self.region)
+
+    @cached_property
+    def awslambda(self):
+        return boto3.client("lambda", self.region)
+
+    @cached_property
+    def ssm(self):
+        return boto3.client("ssm", self.region)
+
+    @cached_property
+    def stepfunctions(self):
+        return boto3.client("stepfunctions", self.region)
+
+    def asg(self, group_names: list[str]):
+        if not group_names:
+            return
+        p = self.autoscaling.get_paginator("describe_auto_scaling_groups")
+        existing_groups = [
+            asg["AutoScalingGroupName"]
+            for i in p.paginate()
+            for asg in i["AutoScalingGroups"]
+            if asg["AutoScalingGroupName"] in group_names
+        ]
+
+        if existing_groups:
+            pprint({"Auto scaling groups to delete": existing_groups})
+
+            if self.delete:
+                for group in existing_groups:
+                    print(self.autoscaling.delete_auto_scaling_group(AutoScalingGroupName=group))
+
+    def eip(self, eip_addresses: list[str]):
+        if not eip_addresses:
+            return
+        result = self.ec2.describe_addresses()
```

Comment:
Doesn't need to be paginated?

Reply:
I'll double check, but I'm pretty sure this was one of the things where the api actually returns everything.

Reply:
Yeah, there's no paging for this call.